packages/std/src/math.rs

```diff
@@ -5,26 +5,43 @@ use std::{fmt, ops};
 
 use crate::errors::{generic_err, underflow, StdError, StdResult};
 
-/// Decimal9 represents a fixed-point decimal value with 9 fractional digits.
-/// That is Decimal9(1_000_000_000) == 1
+/// A fixed-point decimal value with 18 fractional digits, i.e. Decimal(1_000_000_000_000_000_000) == 1.0
 #[derive(Serialize, Deserialize, Copy, Clone, Debug, PartialEq, JsonSchema)]
-pub struct Decimal9(pub u64);
+pub struct Decimal(Uint128);
 
-pub const DECIMAL_FRACTIONAL: u64 = 1_000_000_000;
+const DECIMAL_FRACTIONAL: Uint128 = Uint128(1_000_000_000_000_000_000);
 
-impl Decimal9 {
-    pub fn one() -> Decimal9 {
-        Decimal9(DECIMAL_FRACTIONAL)
+impl Decimal {
+    /// Create a 1.0 Decimal
+    pub const fn one() -> Decimal {
+        Decimal(DECIMAL_FRACTIONAL)
     }
 
-    // convert integer % into Billionth units
-    pub fn percent(percent: u64) -> Decimal9 {
-        Decimal9(percent * 10_000_000)
+    /// Create a 0.0 Decimal
+    pub const fn zero() -> Decimal {
+        Decimal(Uint128(0))
     }
 
-    // convert permille (1/1000) into Billionth units
-    pub fn permille(permille: u64) -> Decimal9 {
-        Decimal9(permille * 1_000_000)
+    /// Convert x% into Decimal
+    pub fn percent(x: u64) -> Decimal {
+        Decimal(Uint128((x as u128) * 10_000_000_000_000_000))
+    }
+
+    /// Convert permille (x/1000) into Decimal
+    pub fn permille(x: u64) -> Decimal {
+        Decimal(Uint128((x as u128) * 1_000_000_000_000_000))
+    }
+
+    pub fn is_zero(&self) -> bool {
+        self.0.u128() == 0
+    }
+}
+
+impl ops::Add for Decimal {
```

Comment:
Maybe `Mul` would be nice... like `Decimal * u64`
So we can do: `Decimal::one() * 5` to make 5.

No need for `Decimal * Decimal` yet.